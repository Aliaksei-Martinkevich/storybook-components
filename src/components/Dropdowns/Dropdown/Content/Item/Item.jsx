import React from 'react';
import PropTypes from 'prop-types';

const Item = ({ onClick, children, id, className }) => (
  <div className={className} onClick={() => onClick(id)}>
    {children}
  </div>
  );

Item.defaultProps = {
  className: '',
};

Item.propTypes = {
  onClick: PropTypes.func.isRequired,
  id: PropTypes.string.isRequired,
  children: PropTypes.element.isRequired,
  className: PropTypes.string,
};

export default Item;
