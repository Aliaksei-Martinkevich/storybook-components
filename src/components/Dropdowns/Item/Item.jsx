import React from 'react';
import PropTypes from 'prop-types';

const Item = ({ onClick, children, id }) => (
  <div onClick={() => onClick(id)}>
    {children}
  </div>
  );

Item.propTypes = {
  onClick: PropTypes.func,
  id: PropTypes.string,
};

export default Item;
