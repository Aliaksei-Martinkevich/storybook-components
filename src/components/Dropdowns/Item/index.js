import React from 'react';
import PropTypes from 'prop-types';

const Item = ({ onClick, children}) => {
  return (
    <div onClick={onClick}>
      {children}
    </div>
  )
};

Item.propTypes = {
  onClick: PropTypes.func,
};

export default Item;
