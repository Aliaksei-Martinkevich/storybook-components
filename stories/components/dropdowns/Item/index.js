import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import Components from '../../../../src/components'

const Item = Components.Dropdowns.Item;

export default storiesOf('Dropdown Item', module)
  .add('With text', () =>
    <Item onClick={action('Item clicked')}>
      Some text
    </Item>);