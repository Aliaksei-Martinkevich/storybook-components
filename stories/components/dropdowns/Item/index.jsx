import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import Item from '../../../../src/components/Dropdowns/Dropdown/Content/Item/Item.jsx';

export default storiesOf('Dropdown Item', module)
  .add('With text', () =>
    (<Item onClick={action('Item clicked')}>
      Some text
    </Item>));
