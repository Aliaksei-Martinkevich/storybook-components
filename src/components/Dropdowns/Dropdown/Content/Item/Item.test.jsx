import React from 'react';
import renderer from 'react-test-renderer';
import Item from './Item.jsx';

describe('Item', () => {
  it('Should render without fails', () => {
    const customClassName = 'class-name-example';
    const component = renderer.create(
      <Item
        className={customClassName}
      />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should send it\'s id into onClick callback', () => {
    const customClassName = 'class-name-example';
    const id = 'sampleId';
    const onClick = jest.fn();
    const component = renderer.create(
      <Item
        id={id}
        onClick={onClick}
        className={customClassName}
      />);
    let tree = component.toJSON();
    tree.props.onClick();
    tree = component.toJSON();
    expect(onClick.mock.calls).toEqual([[id]]);
    expect(tree).toMatchSnapshot();
  });
});
