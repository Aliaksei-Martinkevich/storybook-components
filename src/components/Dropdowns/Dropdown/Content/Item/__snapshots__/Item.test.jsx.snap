// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item Should render without fails 1`] = `
<div
  className="class-name-example"
  onClick={[Function]}
/>
`;

exports[`Item Should send it's id into onClick callback 1`] = `
<div
  className="class-name-example"
  onClick={[Function]}
/>
`;
